### DISCUSSION TITLE
第 5 章 颜色和辐射度学 公式(5.18)

### DISCUSSION THREAD

**Discussion by xwc2021** [kanition/pbrtbook#137] [交流]
對L和I做了比較
發現如果做這樣調整，在幾何上似乎可以更直觀理解L

![擷取](https://github.com/kanition/pbrtbook/assets/11153808/c019c473-ef8b-49d5-abd0-f4e84363b30c)

![L_I](https://github.com/kanition/pbrtbook/assets/11153808/34acea4b-6206-4d3b-a559-7a40c00b3f51)

**Comment by kanition** [comment]
抱歉，我不同意这样的做法。余弦角被你挪到强度I的计算中，就不再是学界公认的强度定义了。这样你在参考其他材料时可能会让自己更加混乱。建议还是先吃透公认的定义，争取跨过这个坎。如之前所说，这可能需要你先增强对相关数学定义的理解。

**Reply by xwc2021** [reply]
那是因為 I 不再是四面八方可以輻射呀，如果用 I =Φ  / 4π ，反而幾何上會有點奇怪不是嗎？

又如果是用 I =Φ  / 4π，那平分給所有平面時，去那裡找cos項呢？ 

![3GX-480x480](https://github.com/kanition/pbrtbook/assets/11153808/0c6b22cc-b0cf-4aa6-8e9a-bf0d5a7fb32c)

**Reply by xwc2021** [reply]
我明白了，你那張圖說的是射入 (能量射入dA)
而我那張圖說的是射出(如何產生光源的L)

以射入來說
你那張圖完全沒問題
很抱歉是我沒弄明白你的意思

但如果是射出，我覺得是要用我改過的那張
![擷取](https://github.com/kanition/pbrtbook/assets/11153808/f9e56142-4ffc-47ae-b6b4-5894b0059506)

**Reply by xwc2021** [reply]
不好意思，再打擾了

我發現入射L的情況
你上面說的微分不一樣
cos(θi)項要和dA綁在一起： dA cos(θi) = dA⊥
因為dw 是用 dA⊥算出來的
dw = dA⊥ / (R*R)

所以圖5.21的2個路線在「測量入射」並沒有差別
「投影面積」跟「立體角」
「立體角」跟「投影面積」

![260015321-f9e56142-4ffc-47ae-b6b4-5894b0059506](https://github.com/kanition/pbrtbook/assets/11153808/5fc7097f-4971-4324-a6cb-d8c01fd71d71)



## from PBR3 book
![M3](https://github.com/kanition/pbrtbook/assets/11153808/b59187cb-d3c7-42a3-bc2a-8fab40f65136)
## 我發現上面的理解是錯的，dw不是透過dA⊥算出來的

## 實驗步驟1
![M1](https://github.com/kanition/pbrtbook/assets/11153808/7f5452fc-8e71-4817-aed4-4de27e157d2b)

## 實驗步驟2
![M2](https://github.com/kanition/pbrtbook/assets/11153808/0c32557a-3595-4673-a190-dbf6b3b5eb60)

## 測量結果為
dω是透過光源dA算出來的
dω
dω = dA / R2

拿上面量到的值做下面的運算
d2Φ / ( dA⊥ dω ) 

推導結果：先微分dA⊥、後微分dω
d2Φ / ( dA⊥ dω ) 

= d(dΦ) / (  dA⊥ dω ) 
= d(dΦ / dA⊥) / dω
= dE / dω = Li

推導結果：或是先微分dω、後微分dA⊥
d2Φ / ( dω dA⊥  ) 

= d(dΦ) / ( dω dA⊥ ) 
= d(dΦ / dω) / dA⊥
= dI / dA⊥ = Li

d2Φ = dΦ cos(θi)
d2Φ / dω = dΦ cos(θi) / dω
dI = I cos(θi)

**Comment by kanition** [comment]
这些公式最关键的一点是理解计算时该代入谁的面积、谁的角度、谁的余弦项。拿立方体的例子来说，你没有说明要计算立方体上某一点朝什么方向的亮度，而是毫无察觉地默认为取垂直于立方体表面的方向，这样余弦项刚好等于1，于是看起来像在公式中消失了。而且这个垂直于立方体表面的方向我猜在你的看法里是由立方体中心连接立方体表面得到的，这又是一个错误——你要计算表面上向外辐射的亮度，那这个方向应该是从立方体表面起始连向观察点（也即里面会有余弦项），与立方体中心没有什么关系。这还意味着，你从不同方向观察这个立方体也许会有不一样的亮度（有余弦项的影响、还有立方体朝各个方向的辐射强度是否是均匀的影响），就看你怎么规定它的发光特性了。此外我认为我在该补充章节中对各概念的定义不够清晰，为了让读者读起来简单而尽量缩短句子，容易造成歧义，后面我会更新的。

**Reply by xwc2021** [reply]
「拿立方体的例子来说，你没有说明要计算立方体上某一点朝什么方向的亮度，而是毫无察觉地默认为取垂直于立方体表面的方向…」
我發現是我讓你誤會了，每個dA上都是朝半球方向，向外輻射的
最後每個方向都是
L cos(θ) = dE / dw
移項後L =dE /(dw cos(θ))
L在每個方向就是定值了

這裡我漏了cos項補一下
![260228547-5fc7097f-4971-4324-a6cb-d8c01fd71d71](https://github.com/kanition/pbrtbook/assets/11153808/cdc54d41-740d-44a1-93b0-8fda7e777b66)

「你要计算表面上向外辐射的亮度，那这个方向应该是从立方体表面起始连向观察点（也即里面会有余弦项），与立方体中心没有什么关系。」你說的沒錯，我是這樣想的

**Reply by kanition** [reply]
我收回之前怀疑照度E计算可能有误的话，经过确认，现在的译文中没有错误，E的计算不应代入余弦项，我差点被你绕进去和朗伯余弦定理搞混了。我似乎明白你为什么想不通了，你想要知道怎么推导出亮度不变定理是吧？我后面可以再增加一下相关推导。

**Comment by kanition** [comment]
关于你的余弦角被你挪到强度I的计算，我虽然不同意，但你举出的例子也让我发现了与朗伯余弦定律确实存在矛盾的地方，这样说来我倒是怀疑我E的定义有错误，应该改为把余弦项挪到E的计算中，关于这一点我应再确认下

**Reply by kanition** [reply]
https://wenku.baidu.com/view/88b0a90203d8ce2f0066233c?aggId=4dfae914961ea76e58fafab069dc5022aaea46be&fr=catalogMain_text_ernie_recall_v1%3Awk_recommend_main3&_wkts_=1691855014453&bdQuery=%E8%BE%90%E5%B0%84%E7%85%A7%E5%BA%A6

**Reply by xwc2021** [reply]
### Receive L的部分
我有發現我那個Receive L的量法，量的其實是面光源的Emitted L
所以才會變成 dw = dA⊥ / (R*R)
PBR3 那張圖，才是Receive L的定義，我的是錯的

![Li](https://github.com/kanition/pbrtbook/assets/11153808/593b35f1-71ba-4f3a-a6a9-54269b7de09c)

那它其實在先微分那個也都一樣
![c](https://github.com/kanition/pbrtbook/assets/11153808/4bf88385-aedf-4817-95f7-580ce96b868b)


### Emitted L的部分
我有看到第22頁那張圖了
我後來覺得這cos項也不是什麼問題
我把它放進去，最後也等於放外面


![L_I](https://github.com/kanition/pbrtbook/assets/11153808/ac03436c-5158-4c02-b6c8-2751ab41b1ea)


![擷取2](https://github.com/kanition/pbrtbook/assets/11153808/749c9d82-8165-4338-b0a4-adf4bf01d6f4)

這個是我最近才發現的
![d2](https://github.com/kanition/pbrtbook/assets/11153808/aed07f67-236f-442a-b954-32dcfa9c160b)

**Reply by kanition** [reply]
总之强度I和照度E各自的计算都不带余弦，但亮度L要带。另外微分号d并不是乘以一个微小量，它有严格的数学定义，建议详查微积分教材，这样你关于d^2的写法也是有问题的

**Reply by xwc2021** [reply]
的確這個d^2的寫法是我自己猜想的，但即使不用這種方法
 cos項還是可以從 d(dI / dw cos) = d( I ) / cos 往外提不是嗎？

我還發現一件事
之前這張圖你還記得嗎？

### 眼睛看到的Li不隨距離而變
這個測法裡，dw不隨距離而變，dE也不會隨距離而變，dE /dw為定值
![IMG_20220331_153046](https://github.com/kanition/pbrtbook/assets/11153808/766d8e44-b5de-4b27-b97a-4a34cfeda8d7)

**Reply by kanition** [reply]
你提到的亮度L不变的理论（亮度守恒定律）我也会更新在后面的版本中。不过相比于你拍的那张图片里面比较依赖直觉的解释，我还是打算参考查到的数学推导，毕竟我们讨论这么久下来已经足以说明有的东西是很反直觉的，有数学证明更便于澄清问题。强度、照度、亮度的定义我也将后续更新写得更加详细避免歧义；此外我再考虑下是否补充解释下朗伯余弦定理和距离平方反比定理。

**Reply by xwc2021** [reply]
我想我們應該都同意Emitted L的情況
Le 和出射方向無關

但我不確定，如果是Received L的情況
Li 到底是要定義成和入射方向無關(在方個方向都一樣)
Li = dE / dw 

還是和入射方向有關(dw固定時，垂直而下測到的dE⊥最強，所以Li也最強)
Li = dE⊥ / dw

dE cos = dE⊥

**Reply by kanition** [reply]
我现在还没有细想，但直觉上我猜测亮度的计算应该对射入和射出保持内在一致性，这样才是自洽的，不会造成什么矛盾。就像经典物理中力做功是用力矢量与位移矢量求内积，不管是正功还是负功都适用。所以我相信亮度的计算不管发射还是接收，对应变量的关系都是一样的

**Reply by xwc2021** [reply]
你說的沒錯，是我搞錯了符號的意義
dE和入射夾角無關，那 L = dE / dω 自然也和入射夾角無關
和入射夾角有關的是dE⊥，BRDF才和入射夾角有關

- 沿著光行進的方向是dE⊥i
- 能全部擊中shading point dAs的是dEi
- dEi = dE⊥i cos(θi)
- dE⊥i 還沒考慮cos(θi)項，所以會比較強
![another_sum_of_dE2](https://github.com/kanition/pbrtbook/assets/11153808/032e2812-28fe-4faa-9402-7a6737d9f05f)


上圖是從已知Emitted L是什麼出發
來看Received L是什麼
- 當光源Emitted L為Lo
- shading point dAs 的Received L為Li
- 多個Lo射向dAs的結果為 Lo dω = dE⊥ ①  
- 由Received L的定義可知 Li  = dE⊥ / dω ②
- ①代入②得到 Li = Lo

我上圖的Li dω 其實是下圖上半部那種情況
![mesurement0](https://github.com/kanition/pbrtbook/assets/11153808/ad8bdf18-7893-4610-a3b9-6c467f35fa65)


從左半部來看
- Li dωi = dE⊥

從右半部來看
- dω是固定的
- 進入dω的dΦ是固定的
- 正面朝向的dA測量到的d2Φ大
- 傾斜朝向的dA測量到的d2Φ小
- 上下2個dE相等 (現實世界測量)

現實世界
- 直接測量到的是d2Φ、而不是dΦ
- L的偏微分定義，可以看成是對d2Φ進行2次除法
- 除dω、還有除dA cos(θi)

**Comment by kanition** [comment]
@xwc2021 本作v0.7.2版已更新5.9.1节内容，请参考

**Reply by xwc2021** [reply]
感謝 🎉

**Reply by xwc2021** [reply]
抱歉給你添麻煩了
看了你新版本，再回頭看舊版
發現你舊版本，本來就沒問題
是我自己當時還認不出來
![260228547-5fc7097f-4971-4324-a6cb-d8c01fd71d71](https://github.com/kanition/pbrtbook/assets/11153808/3faa4b82-8eaa-4e97-9d5e-c4f4876b18b4)
